prometheus_es_exporter/__init__.py

```diff
@@ -194,7 +196,9 @@ def collect(self):
         # as it may be updated by other threads.
         # (only first level - lower levels are replaced
         # wholesale, so don't worry about them)
+        METRICS_BY_QUERY_LOCK.acquire()
         query_metrics = METRICS_BY_QUERY.copy()
+        METRICS_BY_QUERY_LOCK.release()
```

Comment:
I'm a bit torn on whether we actually need to include any locking, and lean towards not including it unless issues are found.

`METRICS_BY_QUERY` is already accessed by multiple threads, as `collect()` is called from the webserver started by `start_http_server()`, which doesn't handle requests in the main thread.

Each key is only ever updated by a single query, so therefore a single thread. As [dictionary assignments are atomic](https://docs.python.org/3/faq/library.html#what-kinds-of-global-value-mutation-are-thread-safe), I'm don't believe any meaningful race conditions are possible.

Reply:
Did you run into any issues without this locking in place in your testing?

Reply:
I didn't, but I didn't test extensively. 

For some context, my team at work wants to use this for an internal feature - we're currently only testing against ~5 queries at a time, but can quickly see it scaling to >100 queries. I didn't realize dictionary assignments were atomic in Python, though, so I agree with you!